src/int.rs

```diff
@@ -0,0 +1,1409 @@
+//! This module contains an 256-bit signed integer implementation.
+
+use crate::errors::{ParseI256Error, TryFromBigIntError};
+use serde::{Deserialize, Serialize};
+use std::cmp;
+use std::convert::{TryFrom, TryInto};
+use std::fmt;
+use std::iter;
+use std::ops;
+use std::str;
+use std::{i128, i64, u64};
+use web3::contract::{self, tokens};
+use web3::types::U256;
+
+/// Compute the two's complement of a U256.
+fn twos_complement(u: U256) -> U256 {
+    let (twos_complement, _) = (!u).overflowing_add(U256::one());
+    twos_complement
+}
```

Comment:
This is so interesting! Today I learned,

https://en.wikipedia.org/wiki/Two%27s_complement

After studying pure mathematics for decades, there are still so many cute little tricks used in computer science that I have never seen before. This file alone contains several of them.